crates/oxc_transformer/src/regexp/mod.rs

```diff
@@ -0,0 +1,246 @@
+        if !(self.options.named_capture_groups
+            || self.options.unicode_property_escapes
+            || self.options.look_behind_assertions)
+        {
+            return false;
+        }
+
+        pattern.body.body.iter().any(|alternative| {
+            alternative.body.iter().any(|element| match element {
+                Term::CapturingGroup(_) if self.options.named_capture_groups => true,
+                Term::UnicodePropertyEscape(_) if self.options.unicode_property_escapes => true,
+                Term::LookAroundAssertion(assertion)
+                    if self.options.look_behind_assertions
+                        && matches!(
+                            assertion.kind,
+                            LookAroundAssertionKind::Lookbehind
+                                | LookAroundAssertionKind::NegativeLookbehind
+                        ) =>
+                {
+                    true
+                }
+                _ => false,
+            })
+        })
+    }
+
+    /// Check if the regular expression contains any unsupported syntax.
+    ///
+    /// Port from [esbuild](https://github.com/evanw/esbuild/blob/332727499e62315cff4ecaff9fa8b86336555e46/internal/js_parser/js_parser.go#L12667-L12800)
+    fn has_unsupported_regex_syntax_raw(&self, pattern: &str, flags: RegExpFlags) -> bool {
+        // Early return if no unsupported features-related plugins are enabled
+        if !(self.options.named_capture_groups
+            || self.options.unicode_property_escapes
+            || self.options.look_behind_assertions)
+        {
+            return false;
+        }
+
+        let is_unicode = flags.contains(RegExpFlags::U);
+        let mut paren_depth = 0;
+        let mut chars = pattern.chars().peekable();
+
+        while let Some(c) = chars.next() {
+            match c {
+                '[' => {
+                    while let Some(c) = chars.next() {
+                        if c == ']' {
+                            break;
+                        }
+                        if c == '\\' {
+                            chars.next();
+                        }
+                    }
+                }
+                '(' => {
+                    if matches!(chars.peek(), Some(&'?')) {
+                        chars.next();
+                        if matches!(chars.peek(), Some(&'<')) {
+                            chars.next();
+                            match chars.peek() {
+                                Some(&'!' | &'=') if self.options.look_behind_assertions => {
+                                    // (?<=) and (?<!)
+                                    return true;
+                                }
+                                _ => {
+                                    if self.options.named_capture_groups {
+                                        return chars.any(|c| c == '>');
+                                    }
```

Comment:
I'm not 100% sure, but is it possible this exits early when it shouldn't in some cases?

```suggestion
                                    if self.options.named_capture_groups && chars.any(|c| c == '>') {
                                        return true;
                                    }
```

Or maybe, since we can assume the RegExp is valid, skip checking for the `>` entirely:

```suggestion
                                    if self.options.named_capture_groups {
                                        return true;
                                    }
```

Reply:
The first one is esbuild way, so I apply it